secure_message/application.py

```diff
@@ -85,6 +94,66 @@ def handle_exception(error):  # NOQA pylint:disable=unused-variable
     return app
 
 
+def cache_client_token(app):
+    token = get_client_token(app.config['CLIENT_ID'],
+                             app.config['CLIENT_SECRET'],
+                             app.config['UAA_URL'])
+
+    r = redis.StrictRedis(host=app.config['REDIS_HOST'],
+                          port=app.config['REDIS_PORT'],
+                          db=app.config['REDIS_DB'])
+    put_token(r, token)
+
+def put_token(conn, token):
+    try:
+        conn.setex('secure-message-client-token', token.get('expires_in') - 15, token)
+    except redis.exceptions.RedisError:
+        logger.exception("RedisError occurred. Retrying.")
+        sleep(0.5)
+        put_token(conn, token)
```

Comment:
If this errors would it get stuck in a loop?

Reply:
Yep